pkgs/by-name/li/libvmi/package.nix

```diff
@@ -0,0 +1,97 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  unstableGitUpdater,
+  autoreconfHook,
+  autoconf-archive,
+  bison,
+  flex,
+  glib,
+  pkg-config,
+  json_c,
+  libvirt,
+
+  withVMIFS ? true,
+  fuse,
+
+  legacyKVM ? false,
+  libkvmi,
+
+  xenSupport ? true,
+  xen-slim,
+}:
+
+let
+  pname = "libvmi";
+  version = "0.14.0-unstable-2024-08-06";
+  libVersion = "0.0.15";
+
+  src = fetchFromGitHub {
+    owner = "libvmi";
+    repo = "libvmi";
+    rev = "bdb9ffb8f1f70b425454bc41da2be353cc6cbf5c";
+    hash = "sha256-5K+9Qq5vGeYYy8kqWIeO25iNJoD/HvtyircH6odr/qA=";
+  };
+in
+
+stdenv.mkDerivation {
+  inherit pname version src;
+
+  outputs = [
+    "out"
+    "lib"
+    "dev"
+  ];
+
+  nativeBuildInputs = [
+    autoreconfHook
+    autoconf-archive
+    bison
+    flex
+    pkg-config
+  ];
+
+  buildInputs =
+    [
+      glib
+      json_c
+      libvirt
+    ]
+    ++ lib.optional xenSupport xen-slim
+    ++ lib.optional (!legacyKVM) libkvmi
+    ++ lib.optional withVMIFS fuse;
+
+  configureFlags =
+    lib.optional (!xenSupport) "--disable-xen"
+    ++ lib.optional legacyKVM "--enable-kvm-legacy"
+    ++ lib.optional withVMIFS "--enable-vmifs";
```

Comment:
```suggestion
    ++ lib.optionals xenSupport [ xen-slim ]
    ++ lib.optionals (!legacyKVM) [ libkvmi ]
    ++ lib.optionals withVMIFS [ fuse ];

  configureFlags =
    lib.optionals (!xenSupport) [ "--disable-xen" ]
    ++ lib.optionals legacyKVM [ "--enable-kvm-legacy" ]
    ++ lib.optionals withVMIFS [ "--enable-vmifs" ];
```

Reply:
No problem, but are you sure about this? They're still singleton lists, and this seems exactly what `lib.optional` is for.

Reply:
For multiple reasons:

1) For helping newcomers reading this code; `lib.optional` wraps the value in a list implicitly, while `lib.optionals` does it explicitly.
2) `configureFlags` is a list, therefore, it makes more sense to specify the brackets explicitly since we're using `++` too.
3) A singleton list using `lib.optional` makes it harder to have multiple items in the list (it could happen in the future). Therefore, using `lib.optionals` makes those changes easier and more straightforward.

Reply:
Thanks. Will make the change tomorrow; it's pretty late right now.

Reply:
It doesn't really matter what we use and we have more pressing issues then this. IMO probably nixfmt should take care of this and not we in reviews.

Reply:
Thanks for hanging in there! Changes addressed.